packages/node/src/evm/verification/request-verification.ts

```diff
@@ -46,14 +41,78 @@ export function verifyDesignatedWallets<T>(
       return [[log], updatedRequest];
     }
 
-    const log = logger.pend(
-      'DEBUG',
-      `Request ID:${request.id} is linked to a valid designated wallet:${request.designatedWallet}`
-    );
+    const message = `Request ID:${request.id} is linked to a valid designated wallet:${request.designatedWallet}`;
+    const log = logger.pend('DEBUG', message);
     return [[log], request];
   });
 
   const logs = flatMap(logsWithVerifiedRequests, (r) => r[0]);
   const verifiedRequests = flatMap(logsWithVerifiedRequests, (r) => r[1]);
   return [logs, verifiedRequests];
 }
+
+export function verifyTriggers(
+  apiCalls: ClientRequest<ApiCall>[],
+  triggers: RequestTrigger[],
+  oises: OIS[]
+): LogsData<ClientRequest<ApiCall>[]> {
+  const logsWithVerifiedApiCalls: LogsData<ClientRequest<ApiCall>>[] = apiCalls.map((apiCall) => {
+    if (apiCall.status !== RequestStatus.Pending) {
+      const message = `Trigger verification skipped for Request:${apiCall.id} as it has status:${apiCall.status}`;
+      const log = logger.pend('DEBUG', message);
+      return [[log], apiCall];
+    }
+
+    const trigger = triggers.find((t) => t.endpointId === apiCall.endpointId);
+    // If the request is for an unknown endpointId, the problem is with the requesting client contract
+    if (!trigger) {
+      const message = `Request:${apiCall.id} has no matching endpointId:${apiCall.endpointId} in Airnode config`;
+      const log = logger.pend('WARN', message);
+      const updatedApiCall = {
+        ...apiCall,
+        status: RequestStatus.Errored,
+        errorCode: RequestErrorCode.UnknownEndpoint,
```

Comment:
I think this should be `UnknownEndpointId`

Reply:
Yeah agreed